### DISCUSSION TITLE
Newbie mistake?

### DISCUSSION THREAD

**Discussion by AgentSells** [marcel-licence/esp32_multitrack_looper#8] [Q&A]
Hi there, totally inspired by your work and looking to learn to replicate and adapt for similar projects in my head.

I'm running Arduino 1.8.19 on OSX, Board: "ESP32 Dev Module, Enabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 115200, Verbose"

I've tried following your guide but after compiling and flashing without error the serial monitor just loops the following.

Loading data
Firmware started successfully
Connect to ES8388 codec... 0x00: 0x00
[  1461][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
0x01: 0x00
[  1474][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
[  1477][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
Failed!
[  2500][W][Wire.cpp:204] begin(): Bus already started in Master Mode.
[  2600][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
0x00: 0x00
[  2612][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
0x01: 0x00
[  2625][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
[  2628][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
Failed!
[  3651][W][Wire.cpp:204] begin(): Bus already started in Master Mode.
[  3751][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
0x00: 0x00
[  3763][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
0x01: 0x00
[  3776][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
[  3779][E][Wire.cpp:313] beginTransmission(): Unfinished Repeated Start transaction! Expected requestFrom, not beginTransmission! Clearing...
Failed!

I am clearly missing something. Any help would be appreciated

**Comment by AgentSells** [comment]
okay i got past that issue by defining 

#define ES8388_PIN_SDA 33
#define ES8388_PIN_SCL 32

after following your thread here 

https://github.com/marcel-licence/esp32_basic_synth/issues/74

**Comment by AgentSells** [comment]
now everything looks ok until here

Total PSRAM: 4192091
Free PSRAM: 4015179
done!..
- Free PSRAM: 2988675
done!..
- Free PSRAM: 1962171
done!..
- Free PSRAM: 935667
not enough memory!
ESP.getFreeHeap() 223992
ESP.getMinFreeHeap() 223712
ESP.getHeapSize() 255256
ESP.getMaxAllocHeap() 110580
Total heap: 255256
Free heap: 223992
Total PSRAM: 4192055
Free PSRAM: 935667
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400d35a3  PS      : 0x00060130  A0      : 0x800d3811  A1      : 0x3ffcb980  
A2      : 0x3ffc394c  A3      : 0x3ffc388c  A4      : 0x3ffc3688  A5      : 0x3ffc3ab4  
A6      : 0x3ffc3aa4  A7      : 0x3ffc3ad8  A8      : 0x00000000  A9      : 0x0000000c  
A10     : 0x3ffc3694  A11     : 0x00000003  A12     : 0x00000000  A13     : 0x00000000  
A14     : 0x00000004  A15     : 0x000000ff  SAR     : 0x0000001e  EXCCAUSE: 0x0000001d  
EXCVADDR: 0x00000000  LBEG    : 0x400d358c  LEND    : 0x400d3650  LCOUNT  : 0x00000000  


Backtrace:0x400d35a0:0x3ffcb9800x400d380e:0x3ffcb9b0 0x400d385f:0x3ffcb9f0 0x400d6c39:0x3ffcba10 




ELF file SHA256: 0000000000000000

Where it now reboots

I saw another discussion mention not enough PSRAM, so i will follow that from here.

**Comment by AgentSells** [comment]
okay i reduced the MAX_LOOP and that fixed it

moving on to trying to setup the same serial monitor as you had in your video. I can't use terraterm as i am on OSX :-(

**Reply by marcel-licence** [reply]
Thank you for the information. If you try to look for an VT100 compatible terminal software it should be able to generate the same output as teraterm.

**Reply by marcel-licence** [reply]
Which version of the ESP32 board library do you use?

**Reply by AgentSells** [reply]
sorry been away from the project for the last week.

Still haven't found any decent terminal software for OSX yet. 

But I know it is working on a basic level as I get audio passthrough and the metronome click when I test it. Troubleshooting the next steps without the monitor is tricky.

Ideally, I want to use physical buttons instead of midi to control it.

I'm running the ESP32-A1s 

looks like this
https://www.amazon.com/Rakstore-ESP32-Audio-Kit-ESP32-A1S-Development-Dual-core/dp/B09BM1C6S5
as far as I can tell this is the data sheet for is 
https://docs.ai-thinker.com/_media/esp32-a1s_v2.3_specification.pdf

**Reply by marcel-licence** [reply]
No problem.

I found some terminals listed here: https://pbxbook.com/other/mac-ser.html 
MacWise looks promising supporting VT100. It may work for you.

There are some little variations of the Audio Kit out there. Having audio is a very good step forward.

I would recommend to apply the little modification if you want to use the buttons to use them with only one GPIO pin. Otherwise you might have some IO conflicts.
